pkg/controller/replication/replication_controller.go

```diff
@@ -456,6 +507,50 @@ func (rm *ReplicationManager) worker() {
 	}
 }
 
+func (rm *ReplicationManager) removeControllerRef(task abandon) error {
+	// We need to get the latest pod from the API server because rc manager may
+	// have taken control of the pod, but the update hasn't been observed
+	// through watch yet. If we ignore the pod, the pod will stuck with the
+	// stale controller ref.
+	// TODO: ControlActivePods is running in a different thread, so it's still
+	// possible to falsely take control of a pod.
+	pod, err := rm.kubeClient.Core().Pods(task.podNamespace).Get(task.podName)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// ignore the task if the pod doesn't exist anymore.
+			return nil
+		}
+		return err
+	}
+	for _, owner := range pod.OwnerReferences {
+		if owner.UID == task.rcUID && owner.Controller != nil && *owner.Controller {
```

Comment:
you're making this stateful by passing in a specific task/UID/etc. The rc manager is currently not stateful (except for expectations which are recomputed periodically). Every rc is just a string passed to the sync routing, that looks up all state information *within* the sync. This makes reasoning about it a lot easier (this is a queue, what happens if we get 2 elements one after the other that fight? I'm not convinced that with quickly changing labels the last applied "abandon" operation is the correct one, are you? even if you are, this is the sort of reasoning problem I'm indicating). 

Can you do something similar and figure out if a pod needs a controller ref or not, without passing args in?